src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_RADIUS = 10;
+    private static final Color DEFAULT_COLOR = Color.WHITE;
+    private static final int MAX_LENGTH = 20;
+    private static final int MIN_LENGTH = 1;
+
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR.name(), DEFAULT_RADIUS);
+    }
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+
+        Figure[] figures = new Figure[]{
```

Comment:
You always generate all Figures, but you only need one
try to use the next example:
```
    public Figure getRandomFigure() {
        int index = random.nextInt(NUMBER_OF_FIGURES);
        switch (index) {
            case 1:
                return prepareCircle();
            case 2:
                return prepareIsoscelesTrapezoid();
            case 3:
                return prepareRightTriangle();
            case 4:
                return prepareRectangle();
            case 5:
                return prepareSquare();
            default:
                return getDefaultFigure();
        }
    }
```